taipy/core/_entity/submittable.py

```diff
@@ -81,13 +82,22 @@ def get_intermediate(self) -> Set[DataNode]:
         all_data_nodes_in_dag = {node for node in dag.nodes if isinstance(node, DataNode)}
         return all_data_nodes_in_dag - self.__get_inputs(dag) - self.__get_outputs(dag)
 
-    def is_ready_to_run(self) -> bool:
+    def is_ready_to_run(self) -> Reason:
         """Indicate if the entity is ready to be run.
 
         Returns:
+            A Reason object that can function as a Boolean value.
             True if the given entity is ready to be run. False otherwise.
```

Comment:
Just want to make it clearer
```suggestion
            A Reason object that can function as a Boolean value,
            which is True if the given entity is ready to be run or there is no reason to be blocked, False otherwise.
```